hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncTableImpl.java

```diff
@@ -822,7 +822,30 @@ private <S, R> void onLocateComplete(Function<RpcChannel, S> stubMaker,
         } else {
           callback.onRegionComplete(region, r);
         }
-        if (unfinishedRequest.decrementAndGet() == 0 && locateFinished.get()) {
+
+        boolean complete = unfinishedRequest.decrementAndGet() == 0 && locateFinished.get();
```

Comment:
The order is a bit strange here...

I think we should first check whether we still need to send a new request to the same region? If so, we just do not need to decrement the unfinishedRequest, as we will send a new request again.